### PR TITLE
Consolidate schema directory trees and related

### PR DESCRIPTION
**Pull request by jpslav**
* Consolidates the `datum/dsl/org/...` and `datum/schema/org/...` directory trees into `schemas/org/...`
* Replaced the avro:generate rake task with our own rake task (copied the original, simplified it, and made it work for our folder structure).  I actually don't think it was working as is.  It was looking in an `avro/dsl` folder which didn't exist -- it didn't complain but it wasn't doing anything. 
* Avro Builder DSL files are now named `avro_builder.rb` (and ilve in e.g. `schemas/org/openstax/ec/nudged/v1`)
* Swagger Blocks event files are now named `swagger.rb` (and live in e.g. `schemas/org/openstax/ec/nudged/v1` and holds a class named e.g. `Ec::Nudged::V1::Swagger`).
* The problem of `class Ec::Nudged::V1::Swagger` giving "Unknown constant 'Ec'" reemerged.  Ruby doesn't know that `Ec` is a module from this definition.  So going forward let's define all of these classes like `module Ec::Nudged::V1\n class Swagger`.

TODOs for @steveburkett 
* [ ] I went ahead and added the `nudged_v1` stuff in the specs and in the v0 bindings extensions so that the tests would pass, but it is to you to see that I did that like we want and if not to fix.  Notably, the swagger definition for nudged still has a version field in it, which is now probably moot.
* [ ] I did not change all the "datum" stuff in the README.
* [ ] There is still "datum" stuff in some swagger descriptions.

Also @steveburkett from now on please generate the model bindings using swagger-codegen in the Docker container, e.g. `./docker/compose run --rm api rake openstax_swagger:generate_model_bindings[X]` so that the version number isn't constantly changing in the generated files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,7 +85,7 @@ curl -X DELETE http://localhost:8083/connectors/datagen-pageviews
 The schema registry is used to encode/decode events stored in kafka.  This repository stores the schemas from the 'ec' namespace.  To modify the schemas, change the schema dsl in 'avro/dsl/*', regenerate the avsc files thru the following rake task
 ```
 bundle exec rake avro:generate
-```
```
Comment (jpslav) [inline README.md:86]: @steveburkett please update your editor to remove blank spaces at the end of lines

File: Rakefile [hunk 2]
```diff
@@ -1,6 +1,20 @@
 # Add your own tasks in files placed in services/tasks ending in .rake,
 # for example services/tasks/capistrano.rake, and they will automatically be available to Rake.
 
+file 'config/secrets.yml' => 'config/secrets.yml.example' do |task|
+  cp task.prerequisites.first, task.name
+end
+
+Rake::Task['config/secrets.yml'].invoke
```
Comment (jpslav) [inline Rakefile:8]: This file was not always being made in Docker runs (for some reason), so added it here.

File: app/bindings/api/v0/bindings/event.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 OpenAPI spec version: 0.1.0
 
 Generated by: https://github.com/swagger-api/swagger-codegen.git
-Swagger Codegen version: 2.4.9
+Swagger Codegen version: 2.4.13
```
Comment (jpslav) [inline app/bindings/api/v0/bindings/event.rb]: @steveburkett using the generate_model_bindings inside Docker will stop this from flipping back and forth.

File: config/application.rb [hunk 4]
```diff
@@ -23,7 +23,8 @@ class Application < Rails::Application
     config.load_defaults 5.2
 
     config.eager_load_paths << Rails.root.join('lib')
-    config.eager_load_paths << Rails.root.join("datum", "schema", "org", "openstax")
+    config.eager_load_paths << Rails.root.join("schemas", "org", "openstax")
+    config.autoload_paths << Rails.root.join("schemas", "org", "openstax")
```
Comment (jpslav) [inline config/application.rb:27]: eager_load_paths is for production, autoload_paths helps us otherwise.

File: config/puma.rb [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 # Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
-port        ENV.fetch("PORT") { 3000 }
+port        ENV.fetch("PORT") { 3001 }
```
Comment (jpslav) [inline config/puma.rb:12]: To match Docker config

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-from ruby:2.6.5
+from ruby:2.6.6
```
Comment (jpslav) [inline docker/Dockerfile:1]: To match .ruby-version

**Review by steveburkett — APPROVED** [review]

**Pull request merged**